src/main.rs

```diff
@@ -197,12 +197,29 @@ fn actual_main(paths: ForemanPaths) -> ForemanResult<()> {
 
             let mut cache = ToolCache::load(&paths)?;
 
+            let mut accumulated_errors: HashMap<&String, ForemanError> = HashMap::new();
```

Comment:
You could simply collect the errors into a Vec<ForemanErrors> that you build from mapping the iterator with something like [`filter_map`](https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.filter_map). 

I need to look back at the error handling stuff to see if it's better to log here in this function or if we already log errors from the entry point.